ext/openssl/ossl_asn1.c

```diff
@@ -842,20 +811,22 @@ int_ossl_asn1_decode0_cons(unsigned char **pp, long max_len, long length,
     }
 
     if (tc == sym_UNIVERSAL) {
-	VALUE args[4];
-	if (tag == V_ASN1_SEQUENCE || tag == V_ASN1_SET)
-	    asn1data = rb_obj_alloc(*ossl_asn1_info[tag].klass);
-	else
-	    asn1data = rb_obj_alloc(cASN1Constructive);
-	args[0] = ary;
-	args[1] = INT2NUM(tag);
-	args[2] = Qnil;
-	args[3] = tc;
-	ossl_asn1_initialize(4, args, asn1data);
-    }
+		if (tag == V_ASN1_EOC)
+			asn1data = rb_funcall(cASN1EndOfContent, rb_intern("new"), 0);
```

Comment:
This should be `cASN1Constructive`.

Only sequence, set, and string types may use constructed encoding, so this is actually an error, but let's not change the current behavior in this PR.

Reply:
I'm removing this conditional, as it's not porting anything from main branch, and I forgot the original context.